src/crimson/os/seastore/cache.cc

```diff
@@ -770,6 +770,11 @@ void Cache::commit_replace_extent(
     CachedExtentRef next,
     CachedExtentRef prev)
 {
+  if (next->version != prev->version + 1 &&
+      is_mutated_ool_applied(prev)) {
+    //adjust next's version if prev has been updated by trim_dirty
+    next->version = prev->version + 1;
+  }
```

Comment:
A concern I have is how we handle the extent's version. I suppose there are two ways:
1. Keep increasing the version whenever the extent is mutated. Then, reset to zero when the version runs into the maximum value of the version---this requires modifying replay_delta() because replay_deta() reads full extent if the version is zero.
2. Reset the version to zero when trim_dirty occurs (this PR takes this approach). The only issue here is that we need to handle version mismatch in case trim_dirty is done in the process of the extent mutation (see above lines).

Reply:
It is an independent issue that seastore didn't handle before, when the extent version is overflow.

The simplest solution to me is similar to 1., but reset the version to `1` when overflow, and keep using `0` (define it to be  `EXTENT_VERSION_NULL`) as the special indicator about the fresh extent. There should be minor changes to support this, mostly that the `delta_info_t::pversion` should be assigned from `i->prior_instance` instead of the plain `i->get_version() - 1`.

Reply:
As an independent fix, it's better to split it into a single commit. It is helpful for the iterative reviews / rebase / cherry-pick, etc.